File: agent_management/public/js/sales_invoice_custom.js
frappe.ui.form.on("Sales Invoice", {
    supplier: function(frm) {
        if (frm.doc.supplier) {
            frappe.db.get_doc("Supplier", frm.doc.supplier).then(supplier => {
                frm.set_value("supplier_sales_partner", supplier.sales_partner);
                frm.set_value("supplier_sales_partner_commission_rate", supplier.sales_partner_commission);

                if (frm.doc.total) {
                    frm.set_value(
                        "supplier_sales_partner_commission",
                        frm.doc.total * (supplier.sales_partner_commission || 0) / 100
                    );
                }
            });
        } else {
            frm.set_value("supplier_sales_partner", "");
            frm.set_value("supplier_sales_partner_commission_rate", 0);
            frm.set_value("supplier_sales_partner_commission", 0);
        }
    }
});

File: agent_management/customization/sales_invoice_custom.py
import frappe
from frappe.utils import flt

def set_supplier_sales_partner_si(doc, method):
    
    if doc.supplier:
        supplier = frappe.get_doc("Supplier", doc.supplier)

        commission_rate = flt(supplier.sales_partner_commission)

        doc.supplier_sales_partner = supplier.sales_partner
        doc.supplier_sales_partner_commission_rate = commission_rate

        if doc.total:
            doc.supplier_sales_partner_commission = (
                flt(doc.total) * commission_rate / 100
            )
def sales_invoice_after_submit(doc, method):
    if doc.workflow_state == "Delivered" and not doc.delivered:
        doc.db_set("delivered", 1)
